framework/core/Project/Math.pm

```diff
@@ -101,6 +113,31 @@ sub _layout2 {
     return {src=>$src, test=>$test};
 }
 
+##
+## Converts file encoding from iso-8859-1 to utf-8
+##
+sub convert_file_encoding {
+    @_ == 1 or die $ARG_ERROR;
+    my ($file_name) = @_;
+    if (-e $file_name){
+        rename($file_name, $file_name.".bak");
+        open(OUT, '>'.$file_name) or die $!;
+        my $converted_file = `iconv -f iso-8859-1 -t utf-8 $file_name.bak`;
+        print OUT $converted_file;
+        close(OUT);
+    }
+}
+
+sub _post_checkout {
+    my ($self, $revision_id, $work_dir) = @_;
+    my $vid = $self->{_vcs}->lookup_vid($revision_id);
+
+    # Convert the file encoding of problematic files
+    my $result = determine_layout($self, $revision_id);
+    convert_file_encoding($work_dir."/".$result->{src}."/org/apache/commons/math3/stat/correlation/StorelessBivariateCovariance.java");
+    convert_file_encoding($work_dir."/".$result->{src}."/org/apache/commons/math3/stat/correlation/StorelessCovariance.java");
+}
+
```

Comment:
What change between Java 8 and Java 11 requires this encoding change? Why is it only needed for two files?

Reply:
The unmodified test case is compiled with -source 5.  This is not supported in JDK 11, so it was changed to -source 6.  In JDK 5 an unmappable character for encoding UTF8 is a warning.  In JDK6 it is an error.  This problem occurs in other projects besides Math.  In your comments for the Lang pull request, you suggest moving the convert routine to Utils, I will do that.